model/map/src/main/java/org/keycloak/models/map/group/MapGroupProvider.java

```diff
@@ -129,11 +129,7 @@ public Long getGroupsCount(RealmModel realm, Boolean onlyTopGroups) {
 
     @Override
     public Long getGroupsCountByNameContaining(RealmModel realm, String search) {
-        ModelCriteriaBuilder<GroupModel> mcb = groupStore.createCriteriaBuilder()
-          .compare(SearchableFields.REALM_ID, Operator.EQ, realm.getId())
-          .compare(SearchableFields.NAME, Operator.ILIKE, "%" + search + "%");
-
-        return tx.getCount(withCriteria(mcb));
+        return searchForGroupByNameStream(realm, search, null, null).count();
```

Comment:
Please keep use of `getCount`, for performance reason.

Reply:
@hmlnarik Not sure that is possible. We need this part
```
tx.read(withCriteria(mcb).pagination(firstResult, maxResults, SearchableFields.NAME))
            .map(MapGroupEntity::getId)
            .map(id -> {
                GroupModel groupById = session.groups().getGroupById(realm, id);
                while (Objects.nonNull(groupById.getParentId())) {
                    groupById = session.groups().getGroupById(realm, groupById.getParentId());
                }
                return groupById;
            }).sorted(GroupModel.COMPARE_BY_NAME).distinct();
``` 
from `searchForGroupByNameStream`.

Reply:
You are right, thanks @martin-kanis . That only stresses the need to improve Javadoc.